File: Zadanie1/Python/main.py
def kalkulator():
    try:
        num1 = int(input("Podaj liczbę 1: "))
        num2 = int(input("Podaj liczbę 2: "))
        operacja = input("Podaj operacje (+, -, *, /): ")
    except ValueError:
        print("Błąd: Podano niepoprawne wartośći")
        return

    if operacja == "+":
        print(num1 + num2)
    elif operacja == "-":
        print(num1 - num2)
    elif operacja == "*":
        print(num1 * num2)
    elif operacja == "/":
        if num2 == 0:
            print("Błąd: Dzielenie przez zero")
            return
        print(round(num1 / num2, 2))
    else:
        print("Błąd: Upewnij się że podane dane są poprawne")

def konwerter_temperatur():
    try:
        konwersja = input("Podaj z jakiego systemu konwertujesz(C/F): ")
        temperatura = int(input("Podaj temperature: "))
    except ValueError:
        print("Błąd: Podano niepoprawne wartości")
        return

    if konwersja == "C" or konwersja == "c":
        print(str(round((temperatura * 1.8) + 32, 2)) + " F*")
    else:
        print(str(round((temperatura - 32) / 1.8, 2)) + " C*")

def srednia_ocen():
    try:
        ilosc_ocen = int(input("Podaj ilość wprowadzanych ocen: "))
    except ValueError:
        print("Błąd: Podano niepoprawne wartości")
        return

    srednia = 0

    for x in range(ilosc_ocen):
        try:
            ocena = 0
            while ocena <= 0 or ocena > 6:
                ocena = int(input("Podaj ocene(1-6) nr " + str(x + 1) + ": "))
            srednia += ocena
        except ValueError:
            print("Błąd: Nie podano liczby")
            return

    srednia /= ilosc_ocen

    print("Średnia wynosi: " + str(round(srednia,2)))
    if srednia >= 3:
        print("Uczeń zdał.")
    else:
        print("Uczeń nie zdał")

def main():
    choice = input("Wybierz podzadanie(1 - kalkulator, 2 - konwersja tmeperatur, 3 - srednia ocen, x - wyjscie z programu): ")
    match choice:
        case '1':
            kalkulator()
        case '2':
            konwerter_temperatur()
        case '3':
            srednia_ocen()
        case 'x':
            return
        case 'X':
            return
        case _:
            print("Nie podano numeru podzadania")
    main()

if __name__ == '__main__':
    main()

File: Zadanie1/C#/Main.cs
using System;

public class Program
{
	static void Kalkulator()
	{
		try
		{
			Console.Write("Podaj liczbę 1: ");
			int num1 = int.Parse(Console.ReadLine());

			Console.Write("Podaj liczbę 2: ");
			int num2 = int.Parse(Console.ReadLine());

			Console.Write("Podaj operacje (+, -, *, /): ");
			string operacja = Console.ReadLine();

			if (operacja == "+")
				Console.WriteLine(num1 + num2);
			else if (operacja == "-")
				Console.WriteLine(num1 - num2);
			else if (operacja == "*")
				Console.WriteLine(num1 * num2);
			else if (operacja == "/")
			{
				if (num2 == 0)
				{
					Console.WriteLine("Błąd: Dzielenie przez zero");
					return;
				}
				
				Console.WriteLine((double)num1 / num2); 
			}
			else
				Console.WriteLine("Błąd: Upewnij się że podane dane są poprawne");
		}
		catch (FormatException)
		{
			Console.WriteLine("Błąd: Podano niepoprawne wartości");
		}
	}

	static void KonwerterTemperatur()
	{
		try
		{
			Console.Write("Podaj z jakiego systemu konwertujesz(C/F): ");
			string konwersja = Console.ReadLine();

			Console.Write("Podaj temperature: ");
			int temperatura = int.Parse(Console.ReadLine());

			if (konwersja == "C" || konwersja == "c")
			{
				double wynik = Math.Round((temperatura * 1.8) + 32, 2);
				Console.WriteLine($"{wynik} F*");
			}
			else
			{
				double wynik = Math.Round((temperatura - 32) / 1.8, 2);
				Console.WriteLine($"{wynik} C*");
			}
		}
		catch (FormatException)
		{
			Console.WriteLine("Błąd: Podano niepoprawne wartości");
		}
	}

	static void SredniaOcen()
	{
		try
		{
			Console.Write("Podaj ilość wprowadzanych ocen: ");
			int iloscOcen = int.Parse(Console.ReadLine());

			double srednia = 0;

			for (int x = 0; x < iloscOcen; x++)
			{
				int ocena = 0;
				while (ocena <= 0 || ocena > 6)
				{
					Console.Write($"Podaj ocene(1-6) nr {x + 1}: ");
					ocena = int.Parse(Console.ReadLine());
				}
				srednia += ocena;
			}

			srednia /= iloscOcen;
			Console.WriteLine($"Średnia wynosi: {Math.Round(srednia, 2)}");

			if (srednia >= 3)
				Console.WriteLine("Uczeń zdał.");
			else
				Console.WriteLine("Uczeń nie zdał");
		}
		catch (FormatException)
		{
			Console.WriteLine("Błąd: Podano niepoprawne wartości");
		}
	}

	public static void Main()
	{
	    while (true)
	    {
            Console.Write("Wybierz podzadanie(1 - kalkulator, 2 - konwersja temperatur, 3 - srednia ocen, x - wyjscie z programu): ");
            string choice = Console.ReadLine();

            switch (choice)
            {
                case "1":
                    Kalkulator();
                    break;
                case "2":
                    KonwerterTemperatur();
                    break;
                case "3":
                    SredniaOcen();
                    break;
                case "x":
                case "X":
                    return;
                default:
                    Console.WriteLine("Nie podano numeru podzadania");
                    break;
            }
        }
	}
}